### PR TITLE
Adding service fabric api version 2020-12-01-preview - rebased with master

### PR DESCRIPTION
**Pull request by a-santamaria**
I took the pr here #10230 and rebased with master to fix the conflicts. Also, update the paths and example to reflect the changes we made on the RP.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

I took @alexgshaw pr here #10230 and rebased with master to fix the conflicts. Also, update the paths and example to reflect the changes we made on the RP

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

To initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by a-santamaria** [comment]
I to @alexgshaw  pr here  #10230 and rebased with master to fix the conflicts. Also, update the paths and example to reflect the changes we made on the RP @jkochhar please take a look at the last commit.

**Review by jkochhar — APPROVED** [review]

**Comment by a-santamaria** [comment]
@chiragg4u could you review this pr. I took the changes from #10230 rebased and update the paths and example to reflect the changes we made on the RP

File: specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-12-01-preview/cluster.json [hunk 1]
```diff
@@ -0,0 +1,1734 @@
...
+          "200": {
+            "description": "The operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/ClusterCodeVersionsListResult"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceFabric/clusters/{clusterName}/upgradableVersion/maximum": {
```
Comment (pilor) [inline specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-12-01-preview/cluster.json:465]: These paths aren't correct. POST actions take place either on a named resource or on the provider itself. This path has two segments after the named cluster which is not valid.

File: specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-12-01-preview/cluster.json [hunk 2]
```diff
@@ -0,0 +1,1734 @@
...
+      "post": {
+        "operationId": "UpgradableVersionMaximum_Post",
```
Comment (pilor) [inline specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-12-01-preview/cluster.json:467]: Separating this out into 3 POST actions is excessive (and leads to the path problems you are running into).  Can you re-model this as a single "/clusters/{clusterName}/listUpgradableVersions" POST action that takes an optional request body.  If the body is not provided it returns both the min and max version in the payload.  If a targetVersion is provided in the body then it returns min, max, and the path to the target.  Using the "list" prefix will also allow users to retrieve this information in a template deployment (which seems valuable)

Comment (a-santamaria) [reply specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-12-01-preview/cluster.json]: @pilor thanks for the suggestion. We have changed it to have a single post action listUpgradableVersions

**Review by jkochhar — APPROVED** [review]

File: specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-12-01-preview/cluster.json (hunk 2 above)
Comment (a-santamaria) [reply specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-12-01-preview/cluster.json]: @pilor I made the changes suggested. Can we get this pr merged?

**Comment by a-santamaria** [comment]
This has been sighed off by ARM. @fengzhou-msft could you help merge the pr? or could you point to somebody to help?

**Comment by a-santamaria** [comment]
Hi @yungezz, this has been signed off by arm, could you merge the pr please?

**Comment by yungezz** [comment]
hi @a-santamaria could you pls fix CI task failure by following guideline in description of the PR? thanks.

**Comment by a-santamaria** [comment]
@yungezz fixed. thanks

**Comment by a-santamaria** [comment]
@yungezz the issues are fixed, could you merge the pr please?

**Comment by yungezz** [comment]
hi @msyyc, could you pls help confirm impact to Python SDK? thanks

**Comment by a-santamaria** [comment]
@yungezz @msyyc these are the breaking changes in python sdk [logs](https://portal.azure-devex-tools.com/api/sdk-dl-pub?p=Azure/azure-rest-api-specs/11614/azure-sdk-for-python/azure-mgmt-servicefabric/logs.txt):

[ChangeLog] **Breaking changes**
[ChangeLog] 
[ChangeLog]   - Model ManagedCluster no longer has parameter reverse_proxy_endpoint_port
[ChangeLog]   - Model ManagedCluster no longer has parameter use_test_extension
[ChangeLog]   - Model ManagedClusterUpdateParameters no longer has parameter reverse_proxy_endpoint_port

not sure why it's showing this because I didn't make any changes on these modes that are located in the 2020-01-01-preview version. These changes were done in this pr that is already merged: https://github.com/Azure/azure-rest-api-specs/pull/10728.
Can we merge this pr?

**Comment by yungezz** [comment]
> @yungezz @msyyc these are the breaking changes in python sdk [logs](https://portal.azure-devex-tools.com/api/sdk-dl-pub?p=Azure/azure-rest-api-specs/11614/azure-sdk-for-python/azure-mgmt-servicefabric/logs.txt):
> 
> [ChangeLog] **Breaking changes**
> [ChangeLog]
> [ChangeLog] - Model ManagedCluster no longer has parameter reverse_proxy_endpoint_port
> [ChangeLog] - Model ManagedCluster no longer has parameter use_test_extension
> [ChangeLog] - Model ManagedClusterUpdateParameters no longer has parameter reverse_proxy_endpoint_port
> 
> not sure why it's showing this because I didn't make any changes here for these modes that are located in the 2020-01-01-preview. These changes were done in a previous pr that is already merged: #10728.
> Can we merge this pr?

hi @msyyc, is it ok for Python SDK? I don't have concern here. thanks.

**Comment by a-santamaria** [comment]
@msyyc can you respond to @yungezz so we can merge the pr please?

**Comment by a-santamaria** [comment]
@msyyc has not responeded. @yungezz can we merge? I'm not going to generate sdks for 2020-01-01-preview which is what the breaking changes validation is complaining about

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
hi @a-santamaria I rerun CI and seems more CI failed, could you pls fix CI failure? For Python SDK breaking change, you're right, it's caused by previous PR, I think it's ok.

**Comment by a-santamaria** [comment]
@yungezz I think they should be fixed now, but not all of them have run. Some checks say "Expected — Waiting for status to be reported"

**Comment by a-santamaria** [comment]
@yungezz I re-ran ci tasks. Only swagger APIDocPreview gets stuck. Can we merge, please?

**Comment by a-santamaria** [comment]
@yungezz any update?

**Review by yungezz — APPROVED** [review]

**Pull request merged**